core/src/main/java/org/zenoss/zep/index/impl/BaseEventIndexBackend.java

```diff
@@ -34,12 +38,44 @@
 
     private final Map<String, SS> savedSearches = new ConcurrentHashMap<String, SS>();
 
+    private final LoadingCache<EventFilter, EventTagSeveritiesSet> tagSeveritiesCache;
+
+    private static final Date IMMEDIATELY = new Date(0);
     private static final Logger logger = LoggerFactory.getLogger(EventIndexBackend.class);
 
-    protected BaseEventIndexBackend(Messages messages, TaskScheduler scheduler, UUIDGenerator uuidGenerator) {
+    protected BaseEventIndexBackend(Messages messages, TaskScheduler scheduler, UUIDGenerator uuidGenerator,
+                                    int tagSeverityCacheSize, int tagSeveritiesCacheTTL) {
         this.messages = messages;
         this.scheduler = scheduler;
         this.uuidGenerator = uuidGenerator;
+        this.tagSeveritiesCache = CacheBuilder.
+                newBuilder().
+                maximumSize(tagSeverityCacheSize).
+                refreshAfterWrite(1, TimeUnit.SECONDS).
```

Comment:
Aha - I had the impression from the docs ("... otherwise refreshes will be performed during unrelated cache read and write operations.") that anything eligible for refresh would be refreshed.

Thanks for clarifying.